From cdec549cd7caffe5305358bf64a43a5f4728759d Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <johannes89@ist-einmalig.de>
Date: Sat, 11 Nov 2017 20:04:25 +0100
Subject: [PATCH 1224/1936] Add SaveOSC test to the official tests

Details:

* Fix bug in SaveOSC test that would overwrite the input xmz file
* Add SaveOSC (currently only with Arpeggio 1) test to the official tests
---
 TODO.txt => TODO-default-values.txt |  2 +-
 rtosc                               |  2 +-
 src/Tests/CMakeLists.txt            |  3 +-
 src/Tests/SaveOSC.cpp               |  3 +-
 src/Tests/load_omz.cpp              | 71 -----------------------------
 5 files changed, 6 insertions(+), 75 deletions(-)
 rename TODO.txt => TODO-default-values.txt (89%)
 delete mode 100644 src/Tests/load_omz.cpp

diff --git a/TODO.txt b/TODO-default-values.txt
similarity index 89%
rename from TODO.txt
rename to TODO-default-values.txt
index ba6cf597..0e0bed1a 100644
--- a/TODO.txt
+++ b/TODO-default-values.txt
@@ -1,8 +1,8 @@
 TODOs for default values:
-* iterate correctly over arrays
 * use b (or #?) for bundles, not a
 * code for random port ("init() to random")
 * remove rDefaultMissing if possible
+* auto range-print
 move code:
 ports.cpp => default_values.cpp
 rtosc.cpp => ...cmp.cpp?
diff --git a/rtosc b/rtosc
index 4ef98353..74d15bf8 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 4ef983536a57636cdd51a514f6ab5ace2879a49b
+Subproject commit 74d15bf80421007e8ce11c40b5b84635ee249a9b
diff --git a/src/Tests/CMakeLists.txt b/src/Tests/CMakeLists.txt
index 22e2e675..9ff1a640 100644
--- a/src/Tests/CMakeLists.txt
+++ b/src/Tests/CMakeLists.txt
@@ -69,7 +69,8 @@ target_link_libraries(save-osc
                       zynaddsubfx_core zynaddsubfx_nio
                       zynaddsubfx_gui_bridge
                       ${GUI_LIBRARIES} ${NIO_LIBRARIES} ${AUDIO_LIBRARIES})
-
+#this will be replace with a for loop when the code will get more stable:
+add_test(SaveOsc save-osc ../../../instruments/examples/Arpeggio\ 1.xmz)
 
 #message(STATUS "Plugin Test ${GUI_LIBRARIES} ${NIO_LIBRARIES} ${AUDIO_LIBRARIES}")
 
diff --git a/src/Tests/SaveOSC.cpp b/src/Tests/SaveOSC.cpp
index d1bd31ea..e2fa3ded 100644
--- a/src/Tests/SaveOSC.cpp
+++ b/src/Tests/SaveOSC.cpp
@@ -145,7 +145,8 @@ class SaveOSCTest
                 // There is actually no need to wait for /save_osc, since
                 // we're in the "UI" thread which does the saving itself,
                 // but this gives an example how it works with remote fron-ends
-                rval = timeOutOperation("/save_osc", filename, 1000)
+                // The filename '""' will write the savefile to stdout
+                rval = timeOutOperation("/save_osc", "", 1000)
                      ? EXIT_SUCCESS
                      : EXIT_FAILURE;
             }
diff --git a/src/Tests/load_omz.cpp b/src/Tests/load_omz.cpp
deleted file mode 100644
index 4cafb0b3..00000000
--- a/src/Tests/load_omz.cpp
+++ /dev/null
@@ -1,71 +0,0 @@
-#include "../Misc/Master.h"
-
-#include <cassert>
-
-void check_files_are_equal(const char* lfilename, const char* rfilename)
-{
-    const int BUFFER_SIZE = 16 * 1024;
-
-    std::ifstream lFile(lfilename);
-    std::ifstream rFile(rfilename);
-    assert(lFile.is_open());
-    assert(rFile.is_open());
-
-    char lBuffer[BUFFER_SIZE];
-    char rBuffer[BUFFER_SIZE];
-    do {
-        lFile.read(lBuffer, BUFFER_SIZE);
-        rFile.read(rBuffer, BUFFER_SIZE);
-        numberOfRead = lFile.gcount();//I check the files with the same size
-        assert(numberOfRead == rFile.gcount());
-
-        if (memcmp(lBuffer, rBuffer, numberOfRead) != 0)
-        {
-//            memset(lBuffer,0,numberOfRead);
-//            memset(rBuffer,0,numberOfRead);
-            return false;
-        }
-    } while (lFile.good() || rFile.good());
-}
-
-int main()
-{
-    assert(argc == 2);
-    const char *filename = argv[1];
-
-    const char* tmp_omz = "/tmp/zynaddsubfx_test_master.omz";
-    const char* tmp_xmz = "/tmp/zynaddsubfx_test_master.xmz";
-
-    {
-        Master master;
-
-        int tmp = master.loadXML(filename);
-        if(tmp < 0) {
-            cerr << "ERROR: Could not load master file " << filename
-                 << "." << endl;
-            exit(1);
-        }
-
-        master.saveOSC(tmp_omz);
-    }
-
-    {
-        Master master;
-
-        master.loadOSC(tmp_omz);
-
-        master.saveXML(tmp_xmz);
-        if(tmp < 0) {
-            cerr << "ERROR: Could not save master file " << tmp_xmz
-                 << "." << endl;
-            exit(1);
-        }
-
-    }
-
-    if(check_files_are_equal(filename, tmp_xmz))
-        return 0;
-    else
-        return 1;
-
-}
-- 
2.47.0

